src/Ganeti/HTools/Loader.hs

```diff
@@ -372,29 +373,49 @@ clearDynU cdata@(ClusterData _ _ il _ _) =
   let il2 = Container.map (\ inst -> inst {Instance.util = zeroUtil }) il
   in Ok cdata { cdInstances = il2 }
 
--- | Checks the cluster data for consistency.
-checkData :: Node.List -> Instance.List
-          -> ([String], Node.List)
-checkData nl il =
-    Container.mapAccum
-        (\ msgs node ->
-             let nname = Node.name node
-                 delta_mem = truncate (Node.tMem node)
-                             - Node.nMem node
-                             - Node.fMem node
-                             - nodeImem node il
-                 delta_dsk = truncate (Node.tDsk node)
-                             - Node.fDsk node
-                             - nodeIdsk node il
-                 newn = node `Node.setXmem` delta_mem
-                 umsg1 =
-                   if delta_mem > 512 || delta_dsk > 1024
-                      then printf "node %s is missing %d MB ram \
-                                  \and %d GB disk"
-                                  nname delta_mem (delta_dsk `div` 1024):msgs
-                      else msgs
-             in (umsg1, newn)
-        ) [] nl
+-- | Update cluster data to use static node memory on KVM.
+setStaticKvmNodeMem :: Node.List -- ^ Nodes to update
+                       -> Int -- ^ Static node size
+                       -> Node.List -- ^ Updated nodes
+setStaticKvmNodeMem nl static_node_mem =
+  let updateNM n = if Node.hypervisor n == Just Kvm
```

Comment:
Wouldn't guards be better?
```
let updateNM n =
        | Node.hypervisor n == Just Kvm = { Node.nMem = static_node_mem }
        | otherwise = n
```